clang/docs/DPCPPToolsRFC.md

```diff
@@ -0,0 +1,260 @@
+Discourse topic details:
+	- Category: [Clang Frontend](https://discourse.llvm.org/c/clang/6)
+	- Title: "RFC: Offloading design for SYCL offload kind and SPIR targets"
+	- Tags: sycl
+    
+# Offloading design for SYCL offload kind and SPIR targets
+
+This RFC is intended to discuss proposed changes to compilation flow for
+offloading SYCL kernels specifically to SPIR-based targets. Most of the changes
+will be made in the `clang-linker-wrapper` tool. To be posted to the Clang
+Frontend category.
+
+## Introduction
+
+Traditional device offloading models are completely encapsulated within the
+compiler driver requiring the driver to perform all of the steps required for
+generating the host and device compilation passes. The driver is also
+responsible for initiating any of the link-time processing that occurs for each
+device target.
+
+An updated offloading model uses the new `clang-linker-wrapper` tool. Much of the
+functionality that is performed during the link phase of the offloading
+compilation is removed from the driver and moved to the `clang-linker-wrapper`
+tool.
+
+Below is a general representation of the overall offloading flow that is
+performed during a full compilation from source to final executable. The
+compiler driver is responsible for creating the multi-targeted object and the
+`clang-linker-wrapper` tool is responsible for the general functionality that is
+performed during the link. This compilation is capable of supporting multiple
+device targets. Support for Intel-based device targets relies on SPIR-V code
+generation. AMDGPU and NVPTX devices can be supported without relying on SPIR-V
+code generation, as shown in the figure below.
+
+![High level view of the offloading flow](images/offloadflow.svg)
+
+*Diagram 1: Overall compilation flow*
```

Comment:
@jbrodman and @mdtoguchi

Can you please look at the compiler flow image and the short description above the figure to clearly specify non-SPIRV flows for non-Intel targets?
Please let me know if more details are warranted here.

Thanks